### DISCUSSION TITLE
"Connection reset by peer" when executing Function

### DISCUSSION THREAD

**Discussion by Shadowfita** [appwrite/appwrite#4853] [Q&A]
### 👟 Reproduction steps

When I run my function through appwrite (execute now button or schedule), it fails with the message: "An internal curl error has occurred within the executor! Error Msg: Connection reset by peer" and no further details. The function runs with zero problems locally on my machine.

### 👍 Expected behavior

It should run as expected or provide more verbose errors when running under development

### 👎 Actual Behavior

![image](https://user-images.githubusercontent.com/11067520/207307247-15400421-547b-4c43-b6a7-68f7012fa40c.png)

### 🎲 Appwrite version

Version 1.1.x

### 💻 Operating system

Linux

### 🧱 Your Environment

Fresh install of appwrite

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@Shadowfita could you please try to disable your firewall and test again? If it works with the firewall disabled, a solution would be to add a firewall rule to allow whatever ports Appwrite is running on.

**Comment by stnguyen90** [comment]
@Shadowfita is your function timing out (15 seconds by default)? Perhaps increasing the timeout will help (for troubleshooting).

**Reply by Shadowfita** [reply]
Thanks @stnguyen90 , I'll give the firewall a go. The timeout is set to 120 seconds currently.

**Reply by Shadowfita** [reply]
Hey @stnguyen90 , no changes with firewall tweaks. Ive also increased the timeout to 10 minutes with no luck. It must be hanging somewhere and not logging, although it works fine locally.

**Comment by Shadowfita** [comment]
@stnguyen90 is it possible for a function to store any log information when encountering a curl error? It makes it hard to diagnose when or what it's currently doing when it fails.

**Reply by stnguyen90** [reply]
It typically does, especially for node functions. Does the [starter code](https://appwrite.io/docs/functions#gettingStarted) deploy and run successfully for you?

**Reply by Shadowfita** [reply]
> It typically does, especially for node functions. Does the [starter code](https://appwrite.io/docs/functions#gettingStarted) deploy and run successfully for you?

When my code runs successfully, all of my logs come through correctly, however when it fails with the error above, none of my console.logs are stored in the executions logs tab. It's just blank.

**Reply by Shadowfita** [reply]
> It typically does, especially for node functions. Does the [starter code](https://appwrite.io/docs/functions#gettingStarted) deploy and run successfully for you?

Yep, runs no worries.

**Comment by stnguyen90** [comment]
@Shadowfita would you please check the runtime container's docker logs to see if there's anything helpful?

Also, this type of troubleshooting might be better on your [Discord](https://appwrite.io/discord/) server.

**Comment by PrateekJain1993** [comment]
I am facing similar issue on appwrite cloud. Please let me know what is the problem

**Reply by sgaabdu4** [reply]
Yeah it has happened a few times for me too.

**Comment by stnguyen90** [comment]
If your function runs fine usually, but sometimes has this problem, this might be your problem: https://github.com/appwrite/appwrite/issues/5233.
